### DISCUSSION TITLE
Action button issue

### DISCUSSION THREAD

**Discussion by agungsapp** [Power-Components/livewire-powergrid#1787] [Q&A] [answered]
I had a problem during the first installation with undefined in the action
![image](https://github.com/user-attachments/assets/84d96d60-ec54-482e-a21d-f0bdccbff0db)

Initially I used the bootstrap template and thought it was not compatible with bootstrap, but when I tried changing it to tailwind it was still the same, can anyone help me? I'm sure I've followed all the steps properly.

**Comment by kontas** [comment]
I have the same issue, did you found anything ?

In the exemple they still use the class() method, but in the V6 is no more in the class PowerComponents\LivewirePowerGrid\Button.php script

``public function actions($row): array
    {
        return [
            Button::add('edit-stock')
                ->slot("Edit <strong>{e($row->name)}</strong>")
                ->class('bg-indigo-500 text-white')
                ->dispatch('clickToEdit', ['dishId' => $row->id]),
        ];
    }``

**Reply by agungsapp** [reply, marked as answer]
yeah my problem is solved by putting these two at the top :
import './../../vendor/power-components/livewire-powergrid/dist/powergrid'
import './../../vendor/power-components/livewire-powergrid/dist/bootstrap5.css'

i rearranged the order in resource/app.js so it becomes like this :

`
import './bootstrap';
// powergrid start
import './../../vendor/power-components/livewire-powergrid/dist/powergrid'
import './../../vendor/power-components/livewire-powergrid/dist/bootstrap5.css'
import flatpickr from "flatpickr";
import TomSelect from "tom-select";
window.TomSelect = TomSelect
// powergrid end

import '../../node_modules/boxicons/css/boxicons.css'
import '../../node_modules/boxicons/dist/boxicons.js'
`

**Reply by kontas** [reply]
> yeah my problem is solved by putting these two at the top : import './../../vendor/power-components/livewire-powergrid/dist/powergrid' import './../../vendor/power-components/livewire-powergrid/dist/bootstrap5.css'
> 
> i rearranged the order in resource/app.js so it becomes like this :
> 
> ` import './bootstrap'; // powergrid start import './../../vendor/power-components/livewire-powergrid/dist/powergrid' import './../../vendor/power-components/livewire-powergrid/dist/bootstrap5.css' import flatpickr from "flatpickr"; import TomSelect from "tom-select"; window.TomSelect = TomSelect // powergrid end
> 
> import '../../node_modules/boxicons/css/boxicons.css' import '../../node_modules/boxicons/dist/boxicons.js' `

Glad to see it's solve for you. I have try but i'm using tailwind and i didn't understand that much to fix that :/

**Comment by masirfani** [comment]
![image](https://github.com/user-attachments/assets/2be4dc55-3969-46c3-92b8-7ada46395a2a)
i still got undefined after do thats all

**Comment by eafarooqi** [comment]
I also had this issue, i just recompiled all my assets and clear the cache. Working fine now. 
```
npm run build
```

also cleared browser cache and cookies as well.
